### PR TITLE
Add ansible purge script for hosted-engine.  Completely removes engin…

### PR DESCRIPTION
**Pull request by RabidCicada**
…e and dependency packages to allow reinstall with possible updates

Fixes issue # (delete if not relevant)

Changes proposed in this pull request:

-

-

-

I confirm that this pull request was submitted according to the [contribution guidelines](https://github.com/oVirt/ovirt-site/blob/master/CONTRIBUTING.md): (please @mention yourself to sign)

This pull request needs review by: (please @mention the reviewer if relevant)

### CODE REVIEW

File: source/documentation/how-to/hosted-engine.html.md [hunk 1]
```diff
@@ -346,3 +346,81 @@ If your hosted engine install fails, you have to manually clean up before you ca
...
+          register: ohes_cleanup
+        - name: Debug ohes_cleanup.stdout
+          debug:
+            var: ohes_cleanup.stdout
+        - name: Run Ovirt's Hosted Engine Cleanup Script
+          shell: ovirt-hosted-engine-cleanup -q
+          when: ohes_cleanup.stdout != ""
+        - name: Clean old packages
+          package:
+            name: "{{item}}"
```
Comment (sandrobonazzola) [inline source/documentation/how-to/hosted-engine.html.md]: You should ensure you're not on oVirt Node when doing this.

File: source/documentation/how-to/hosted-engine.html.md [hunk 2]
```diff
@@ -346,3 +346,81 @@ If your hosted engine install fails, you have to manually clean up before you ca
...
+            - "/etc/pki/keystore"
+            - "/etc/ovirt-hosted-engine"
+            - "/var/lib/libvirt/"
+            - "/var/lib/vdsm/"
+            - "/var/lib/ovirt-hosted-engine-*"
+            - "/var/log/ovirt-hosted-engine-setup/"
+            - "/var/cache/libvirt/"
+            - "/etc/libvirt/nwfilter/vdsm-no-mac-spoofing.xml"
+            - "/var/cache/tomcat/temp/*"
+            - "/var/cache/tomcat/work/Catalina/localhost/*"
```
Comment (sandrobonazzola) [inline source/documentation/how-to/hosted-engine.html.md]: Why tomcat?

**Comment by sandrobonazzola** [comment]
@tiraboschi , @irosenzw can you please review?

File: source/documentation/how-to/hosted-engine.html.md (hunk 2 above)
Comment (RabidCicada) [reply source/documentation/how-to/hosted-engine.html.md]: Sorry about that...I thought I cleaned out all the cruft from my specific special software
I must have missed tomcat.

I'll edit one more time.

**Comment by RabidCicada** [comment]
I agree you have a script that does do some of this.  I actually call that script first if it exists on the target system.

In my experience trying to reinstall and reinstall over and over again....other leftover cruft led to install failures of hosted-engine.  This ansible script was the final complete purge solution that allowed everything to work

I need to cleanup some forgotten tomcat stuff and then you can accept or reject as desired.

**Comment by RabidCicada** [comment]
Also...whoa...you guys have updated the ovirt-hosted-engine-cleanup script a lot since I last looked at it.  I see it is much more thorough now.  :).  Perhaps this script is not needed in the general community anymore...but it's existence could be published a little better.

**Comment by gregsheremeta** [comment]
Thanks for your contribution!

> you guys have updated the ovirt-hosted-engine-cleanup script a lot since I last looked at it

Per that and Ido's comment, please help update the tool to provide any missing functionality. Let's not have a second tool. Thanks!!

**Pull request closed without merging**